prime-router/src/main/kotlin/TransportType.kt

```diff
@@ -116,6 +116,8 @@ data class RESTTransportType
     val authTokenUrl: String,
     /** [tlsKeystore]The name for the credential manager to get the JKS used in TLS/SSL */
     val tlsKeystore: String? = null,
+    /** [parameters] The map of parameters to be sent in the message */
+    val parameters: Map<String, String>? = emptyMap(),
```

Comment:
If the intention is that this is never null, the `?` should be removed so that'll be enforced by the compiler.
```suggestion
    val parameters: Map<String, String> = emptyMap(),
```

Reply:
Old REST Transports (i.e NY, OK, WA, .. etc) have null as Parameters.  I intentionally have it for backward compatibility with the old RESTransport.  Currently, only REST Transport for ETOR has parameters.

Reply:
Using the default value `emptyMap()` here means that even for those old transport, the Map should be an empty map, but still not `null`. We should test that to make sure, but given that there is a non-null default value, the only way for this to be `null` would be to have it explicitly set.

Reply:
I set this to default as emptyMap().  If transport doesn't provide the parameters, it will default as an emptyMap().  I am sorry in the earlier response I used null instead of emptyMap().  In the code, it checks for .isEmpty() or not.  I tested a transport with parameters and without parameters and it works.

Reply:
I understand, since it is not intended to ever be null, the `?` should be removed so that will be enforced by the compiler.